presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduSplitManager.java

```diff
@@ -45,10 +50,25 @@ public ConnectorSplitSource getSplits(
             ConnectorTableHandle table,
             SplitSchedulingStrategy splitSchedulingStrategy,
             DynamicFilter dynamicFilter)
+    {
+        long blockingTimeoutMillis = getDynamicFilteringWaitTimeout(session).toMillis();
+        if (blockingTimeoutMillis <= 0 || !dynamicFilter.isAwaitable()) {
+            return getSplitSource(table, splitSchedulingStrategy, dynamicFilter);
+        }
+        // wait narrowing down twice
+        CompletableFuture<ConnectorSplitSource> future = dynamicFilter.isBlocked()
```

Comment:
If I'm understanding this correctly, this will wait for narrowing down of dynamic filter only twice, ideally we want to wait for dynamic filter to complete.

Reply:
Yes, I hope to wait until complete, but did not find a easy way, do you have some sugestions?

Reply:
Modified